lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -778,14 +803,46 @@ def accessible_objects(user_or_attribute, operation, options = {})
       end
     end
 
+    # Returns a map of operation names to the user_or_attribute's accessible objects via each operation
+    def accessible_objects_for_operations(user_or_attribute, operations, options = {})
+      unless options[:direct_only]
+        raise ArgumentError, 'Functionality for indirect objects is not yet implemented!'
+      end
+
+      # convert to operation names if operation instances given
+      operations = operations.map { |o| operation_identifier(o) }
+
+      # default objects for each operation to empty list
+      permitted_map = operations.map { |o| [o, []] }.to_h
```

Comment:
This may be a premature optimization, but is there any opportunity for parallelization here?  Either parallelizing the calls to `objects_for_user_or_attribute_and_operations` and/or parallelizing the permitted vs. prohibitions computation.

Reply:
Yes, we could do threaded calls to `objects_for_user_or_attribute_and_operations` similar to what we added to Dalton priv calculations.  However this second call to `objects_for_user_or_attribute_and_operations` for the prohibitions is only added for completeness here.  Dalton will only ever call this function with `ignore_prohibitions=true` because it has to do the permitted vs prohibited calculations itself due to it storing the right side (object side) of the graph in Neptune.

I have a draft PR open there where you can see this.  I didn't add threading there either because I think when we did add that, it was more trouble that it was worth: lost trace ids on logs, DB connection problems, etc.  Still open for debate though.

Reply:
yea i def don't wanna add threading to this gem. that being said i think we can do this in few queires: if `:ignore_prohibitions` is false i think we can run:

```
     prohibited_and_permitted_map = objects_for_user_or_attribute_and_operations(
        user_or_attribute,
        operations + prohibitions,
        options.except(:filters)
      )
```
this would change the filtering logic after that but would cut down the number of db queries.

all that being said i hear ya that dalton is only gna call it with `ignore_prohibitions=true`

Reply:
ah nice, yea with adding the ops and prohibs together as you suggest we don't even need to consider threading.. perhaps this could be done from dalton also.  guess it is an advantage of building this map.

Reply:
in fact, we could even use this to avoid threading on privilege/show i think

Reply:
oh hell yea i would looove to rip threading out of priv show